### PR TITLE
renaming backend `func_info` dictionary's keys

### PR DESCRIPTION
**Pull request by Schefflera-Arboricola**
Changes made : 
- added `url` : if we would want to link to the function's webpage page(in the future) or source code, right now in nx-parallel I've put the link to the function header(https://github.com/networkx/nx-parallel/pull/47)
-  renamed`extra_parameters` to `additional_parameters`: bcoz "extra" gives a sense that they are not necessary but "additional" gives a sense that they are an added functionality, anyways we can keep it if it's not that big of a change
- renamed `extra_docstring` to `additional_docs` for the similar reason as above
- removed an unused import

### CODE REVIEW

**Comment by MridulS** [comment]
@eriknw here is a proposal to update the variable names for passing the documentation information from the backend to networkx. What do you think?

File: networkx/utils/backends.py [hunk 1]
```diff
@@ -981,6 +994,10 @@ def _make_doc(self):
             else:
                 lines.append("")
 
+            if "backend_func_url" in func_info and func_info["backend_func_url"]:
+                lines.append(f"[`Source <{func_info['backend_func_url']}>`_]")
```
Comment (eriknw) [inline networkx/utils/backends.py:998]: What do you think of using the [walrus operator](https://docs.python.org/3/whatsnew/3.8.html#assignment-expressions) here and elsewhere in this PR? For example:
```suggestion
            if func_url := func_info.get("backend_func_url"):
                lines.append(f"[`Source <{func_url}>`_]")
```

File: networkx/utils/backends.py [hunk 2]
```diff
@@ -960,19 +959,33 @@ def _make_doc(self):
...
+            ):
+                key = (
+                    "extra_docstring"
+                    if func_info.get("extra_docstring")
+                    else "backend_func_docs"
+                )
                 lines.extend(
-                    f"  {line}" if line else line
-                    for line in func_info["extra_docstring"].split("\n")
+                    f"  {line}" if line else line for line in func_info[key].split("\n")
```
Comment (eriknw) [inline networkx/utils/backends.py:973]: And also:
```suggestion
            if func_docs := (
                func_info.get("backend_func_docs") or func_info.get("extra_docstring")
            ):
                lines.extend(
                    f"  {line}" if line else line for line in func_docs.split("\n")
```
(feel free to play around to find syntax that you like)

Comment (Schefflera-Arboricola) [reply networkx/utils/backends.py]: sry I wasn't able to do "commit suggestion" after i pushed some commits.

**Comment by dschult** [comment]
@eriknw could you approve this PR if it is good to go from your perspective? I don't want to mess up anything you are working on that this might touch.

**Comment by eriknw** [comment]
I hadn't considered the connotation difference between "extra" and "additional", but, yeah, let's go with "additional".

Regarding the new names, I think having `"backend_func_*"` is redundant. It is _already_ defined within function information for a backend--specifically, `nx.utils.backends.backend_info[backend_name]["functions"][function_name][key]`.
- So instead of `"backend_func_url"` as `key`, it could simply be `"url"`
  - I like adding this!
- Similarly, `"backend_func_docs"` could be `"doc"` or `"additional_doc"` (`doc`, `docs`, `docstring`?)
  - `"additional_doc*"` matches naming style of `"additional_parameters"`

**Comment by Schefflera-Arboricola** [comment]
Thanks for the review @eriknw ! 

> I hadn't considered the connotation difference between "extra" and "additional", but, yeah, let's go with "additional".

“additional" is used when talking about something that is added to what already exists. And “extra” is used when talking about something that is in addition to what is necessary or required. In the case of backend arguments, it might(or might not) be that some of the backend kwargs are necessary. So, "additional" seemed more appropriate to me.

**Review by dschult — APPROVED** [review]
This looks good to me and Erik has given feedback on the names.
Thanks @Schefflera-Arboricola

**Pull request merged**